src/test/java/uk/gov/ons/ctp/response/collection/exercise/client/ActionSvcClientTest.java

```diff
@@ -105,6 +110,85 @@ public void createActionPlan_500Response() {
         .postForObject(any(URI.class), any(HttpEntity.class), eq(ActionPlanDTO.class));
   }
 
+  @Test
+  public void updateActionPlanNameAndDescription_200Response() throws CTPException {
+
+    // Given
+    when(restTemplate.exchange(
+            any(URI.class), eq(HttpMethod.PUT), any(HttpEntity.class), eq(ActionPlanDTO.class)))
+        .thenReturn(actionPlanResponseEntity);
+
+    // When
+    ActionPlanDTO actionPlan =
+        actionSvcRestClient.updateActionPlanNameAndDescription(
+            UUID.fromString(ACTION_PLAN_ID), "test_name", "test_description");
+
+    assertEquals(actionPlan.getId(), actionPlans.get(0).getId());
+  }
+
+  @Test(expected = HttpClientErrorException.class)
+  public void updateActionPlanNameAndDescription_500Response() {
+    // Given
+    when(restTemplate.exchange(
+            any(URI.class), eq(HttpMethod.PUT), any(HttpEntity.class), eq(ActionPlanDTO.class)))
+        .thenThrow(new HttpClientErrorException(HttpStatus.INTERNAL_SERVER_ERROR));
+
+    // When
+    actionSvcRestClient.updateActionPlanNameAndDescription(
+        UUID.fromString(ACTION_PLAN_ID), "test_name", "test_description");
+
+    // Then
+    verify(restTemplate, times(1))
+        .exchange(any(), eq(HttpMethod.PUT), any(HttpEntity.class), eq(ActionPlanDTO.class));
+  }
+
+  @Test
+  public void getActionPlanById_200Response() throws CTPException {
+
+    // Given
+    when(restTemplate.exchange(
+            any(URI.class), eq(HttpMethod.GET), any(HttpEntity.class), eq(ActionPlanDTO.class)))
+        .thenReturn(actionPlanResponseEntity);
+
+    // When
+    ActionPlanDTO actionPlan =
+        actionSvcRestClient.getActionPlanById(UUID.fromString(ACTION_PLAN_ID));
+
+    assertEquals(actionPlan.getId(), actionPlans.get(0).getId());
+  }
+
+  @Test
+  public void getActionPlanByID_404Response() throws CTPException {
+    // Given
+    when(restTemplate.exchange(
+            any(URI.class), eq(HttpMethod.GET), any(HttpEntity.class), eq(ActionPlanDTO.class)))
+        .thenThrow(new HttpClientErrorException(HttpStatus.NOT_FOUND));
+
+    // When
+    ActionPlanDTO actionPlan =
+        actionSvcRestClient.getActionPlanById(UUID.fromString(ACTION_PLAN_ID));
+
+    // Then
+    verify(restTemplate, times(1))
+        .exchange(any(), eq(HttpMethod.GET), any(HttpEntity.class), eq(ActionPlanDTO.class));
+    assertNull(actionPlan);
+  }
+
+  @Test(expected = HttpClientErrorException.class)
+  public void getActionPlanByID_500Response() {
+    // Given
+    when(restTemplate.exchange(
+            any(URI.class), eq(HttpMethod.GET), any(HttpEntity.class), eq(ActionPlanDTO.class)))
+        .thenThrow(new HttpClientErrorException(HttpStatus.INTERNAL_SERVER_ERROR));
```

Comment:
Same for this test, you'll never get a 500 in a HttpClientErrorException

Reply:
done